src/cljs/athens/devcards/right_sidebar.cljs

```diff
@@ -1,89 +1,219 @@
+                           :max-width "100%"
+                           :overflow "hidden"
+                           :align-items "center"
+                           :color (color :body-text-color)}
+                      [:svg {:opacity (:opacity-med OPACITIES)
+                             :display "inline"
+                             :vertical-align "-4px"
+                             :margin-right "0.2em"}]]
+                     [:.controls {:display "flex"
+                                  :flex "0 0 auto"
+                                  :align-items "stretch"
+                                  :flex-direction "row"
+                                  :transition "opacity 0.3s ease"
+                                  :opacity "0.25"}]
+                     [:&:hover [:.controls {:opacity "1"}]]
+                     [:svg {:font-size "18px"}]
+                     [:hr {:width "1px"
+                           :background (color :panel-color)
+                           :border "0"
+                           :margin "4px"
+                           :flex "0 0 1px"
+                           :height "1em"
+                           :justify-self "stretch"}]
+                     [:&.open [:h2 {:font-weight "500"}]]]})
+
+
+;;; Components
 
 
 (defn right-sidebar-el
-  [state]
-  (let [uids (->> @(q '[:find ?uid ?title
-                        :in $ [?uid ...]
-                        :where
-                        [?e :block/uid ?uid]
-                        [?e :node/title ?title]]
-                      db/dsdb uids)
-                  vec
-                  (reduce-kv
-                    (fn [m _ [uid title]]
-                      (assoc m uid {:open  false
-                                    :title title}))
-                    {}))
-        s (r/atom {:uids uids})]
-    (when (:open state)
-      [:div (use-style right-sidebar-style)
-       [:div (use-style content-style)
-        (doall
-          (for [[uid {:keys [open title]}] (:uids @s)]
-            ^{:key uid}
-            [:div {:style {:display "flex"}}
-             [:span (use-style toggle-page-style
-                      {:on-click (fn [_]
-                                   (swap! s update-in [:uids uid :open] not))})
-              [:> mui-icons/KeyboardArrowDown]]
-             (if open
-               [node-page-component [:block/uid uid]]
-               [:div [:h1 title]])]))]
-       [:div {:on-click (fn [_] (dispatch [:toggle-right-sidebar]))}
-        [:> mui-icons/Close]]])))
+  [open? items]
+  [:div (use-style sidebar-style {:class (if open? "open" "closed")})
+   [:div (use-style sidebar-content-style)
+    [:header (use-style sidebar-section-heading-style)
+     [:h1 "Pages and Blocks"]
+    ;;  [button {:label [:> mui-icons/FilterList]}]
+     ]
+    (doall
+      (for [[uid {:keys [open node/title block/string]}] items
+            :let [node-page? (boolean title)
+                  heading (or title string)]]
```

Comment:
Can remove this `:let` binding entirely. Replace the `if node-page` with `if title`